pkg/util/timeutil/pgdate/field_extract_test.go

```diff
@@ -107,6 +113,50 @@ func TestExtractSentinels(t *testing.T) {
 	}
 }
 
+func TestExtractInvalidJulianDate(t *testing.T) {
+	testCases := []struct {
+		wanted   fieldSet
+		str      string
+		expError string
+	}{
+		// Expect only a time, reject a year.
+		{timeFields, "j69001", `value -4524 for field Year already present or not wanted
+Wanted: [ Hour Minute Second Nanos Meridian TZHour TZMinute TZSecond ]
+Already found in input: [ ]`},
+		// Expect a date, reject when the year is specified twice.
+		{dateFields, "j69001 j69002", `value -4524 for field Year already present or not wanted
+Wanted: [ Era ]
+Already found in input: [ Year Month Day ]`},
+		/// Expect a date+time, reject when the date/month/day is specified twice.
+		{dateTimeFields, "2010-10-10 j69002", `could not parse field: -10
+Wanted: [ Era Nanos Meridian ]
+Already found in input: [ Year Month Day Hour Minute Second TZHour TZMinute TZSecond ]`},
+	}
+
+	// TODO(knz): This would benefit from datadriven testing.
+
+	now := timeutil.Unix(42, 56)
+	for _, tc := range testCases {
+		fe := fieldExtract{
+			currentTime: now,
+			wanted:      tc.wanted,
+		}
+		err := fe.Extract(tc.str)
+		if err == nil {
+			t.Errorf("%+v: expected error, got no error", tc)
+			continue
+		}
+
+		msg := err.Error()
+		msg += "\n" + errors.FlattenDetails(err)
+		if msg != tc.expError {
```

Comment:
are the error messages here stable enough be equality-checked in unit tests?

Reply:
in this case, yes - the full text of the message is generated within this package.
(I wouldn't be comfortable including bits of messages coming from other packages)